### PR TITLE
[CMake] Add build-system vendor hook for swiftCore

### PR DESCRIPTION
**Pull request by etcwilde**
Adding a hook point for vendors to add flags, definitions, etc, to the swiftCore library build.

### CODE REVIEW

**Comment by etcwilde** [comment]
@swift-ci please smoke test

**Review by justice-adams-apple — APPROVED** [review]

File: Runtimes/Core/core/CMakeLists.txt [hunk 1]
```diff
@@ -339,6 +339,8 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/Swift.swiftmodule"
 emit_swift_interface(swiftCore)
 install_swift_interface(swiftCore)
 
+include("${SwiftCore_VENDOR_MODULE_DIR}/swiftCore.cmake" OPTIONAL)
```
Comment (compnerd) [inline Runtimes/Core/core/CMakeLists.txt:342]: what do you think of naming this `VendorCustomizations.cmake`?

Comment (etcwilde) [reply Runtimes/Core/core/CMakeLists.txt:342]: Eventually we will have `_Concurrency` and `Onosupport` libraries built in this project, and we should be able to target changes at those libraries too.
We could put one toward the end of the main `CMakeLists.txt` and use that to target various libraries directly, I suppose?

Comment (compnerd) [reply Runtimes/Core/core/CMakeLists.txt:342]: Oh, wait, I see `SwiftCore_VENDOR_MODULE_DIR` ... so this seems pretty reasonable.

**Review by compnerd — APPROVED** [review]

**Pull request merged**